src/Components/LandingPage/LandingPage.tsx

```diff
@@ -56,7 +57,11 @@ export const LandingPage = () => {
   const [selectedBlueprint, setSelectedBlueprint] = useState<
     string | undefined
   >();
-  const { data: blueprints, isLoading } = useGetBlueprintsQuery({});
+  const [searchQuery, blueprintsFilter, setBlueprintsFilter] =
+    useDebouncedSearch('', 500);
```

Comment:
I won't block this custom hook, but lodash has debounce implementation already, why not using it and keep it as regular state?

```js
import _debounce from 'lodash/debounce';

    const debounceFn = useCallback(_debounce(setBlueprintsFilter, 500), []);
        <SearchInput
             placeholder="Search by name or description"
             value={filter}
             onChange={(_event, value) => debounceFn(value)}
             onClear={() => onChange('')}
```

Reply:
Also, 500 ms feels a bit long to me, would 300 be better? also do we want to add a spinner or another indicator for fetching the data?

Reply:
because I did not find a way how to use lodash debounce :stuck_out_tongue:
Definitelly on looks better, let me improve it! :+1:

Reply:
:)
 I'm thinking about mixing  backend and frontend filtering would work the best together , we already have 100 blueprints records in the store, we can  filter it locally by default for given instant result, and update the store if there were any updates from the debounced query., though I think this enhancement  can  be implemented as a follow up for a general optimistic updating refactor.

Reply:
That's IMO amazing idea, we should discuss it with Andrew Dewar, he did quite a lot of optimistic updating, we can get some lessons learned :)

Reply:
Wow... it's amazing, just 200 ms less and it really feels much smoother experience.
So I've read about it because I was amazed and I've learned that smooth feeling is when the interaction computer <-> human is below 400ms, cool